### ISSUE TITLE
Missing Fields for (lng & lat) at Fixed Location

### ISSUE THREAD

**Issue by Dtrieb** [meshtastic/web#87]
In the new Version the Fields for lng and lat at Fixed Location are missing.

**Comment by garthvh** [comment]
Toggle fixed positon on and they show

**Comment by Dtrieb** [comment]
> Toggle fixed positon on and they show

No in this version, not anymore.

**Comment by dieterthiess** [comment]
![screencapture-client-meshtastic-org-2023-02-13-22_53_43](https://user-images.githubusercontent.com/3475732/218583405-3e0805c8-bdfa-455b-b50a-0248d479721f.png)

**Comment by sachaw** [comment]
Yeah, sorry, currently an omission, I will be adding a modal that lets you enter a fixed position, but I think a change in the protobuds is in order.

**Comment by Hunter275** [comment]
Echoing what @sachaw this is going to take a protobuff change

Current implementation on iOS just takes whatever the last position reported by the GPS was and uses that for the position. You cannot set specific Lat/Lon/Alt

**Comment by OZ2CPU-thomas** [comment]
Why is this closed ?? when it is still not fixed ?
lot long fields DONT show up, when selecting fixed position.
any work arrounds ?

**Comment by Hunter275** [comment]
> Why is this closed ?? when it is still not fixed ? lot long fields DONT show up, when selecting fixed position. any work arrounds ?

This has not been closed

**Comment by Fredozen** [comment]
While waiting for the correction, is it possible to know the settings in the config.yaml file to activate the fixed position and the long/lat/H coordinates so that this is taken into account?

**Comment by pleasantone** [comment]
The CLI command seems to be able to do this?

**Comment by Fredozen** [comment]
Normally, yes, it should be possible. But you need to install the package

**Comment by Fredozen** [comment]
I have a raspberry with a lora hat and obviously the meshtastic client can't communicate with it on an SPI.

**Comment by Fredozen** [comment]
And in the config.yaml file I put:
location:
  lat: 49.xx
  lon: 6.xx
  alt: 210
  fixed: true
And yet I have the message:

[mqtt] MQTT Map report enabled, but no position available

**Comment by Fredozen** [comment]
The only way I've found is to use the Android app to set the position, and it does it correctly.

**Comment by twaddington** [comment]
I have started working on this in https://github.com/twaddington/meshtastic-web/tree/fix/87.

So far I've added the missing fields to the Position Settings page and am pulling in the current position.

From looking at the Android implementation a `setFixedPosition` function needs to be added to the `Device` interface. I'm still working that out.

I also need to fix the types for the Position form fields.

**Comment by dzienisz** [comment]
Can I help you? It's madness it's not working for last 2/3 years.

**Comment by dzienisz** [comment]
> I have started working on this in https://github.com/twaddington/meshtastic-web/tree/fix/87.
> 
> So far I've added the missing fields to the Position Settings page and am pulling in the current position.
> 
> From looking at the Android implementation a `setFixedPosition` function needs to be added to the `Device` interface. I'm still working that out.
> 
> I also need to fix the types for the Position form fields.

Please make a draft PR

**Comment by twaddington** [comment]
@dzienisz I'm chipping away at it. I should have a draft up this week.

**Comment by NellowTCS** [comment]
> [@dzienisz](https://github.com/dzienisz) I'm chipping away at it. I should have a draft up this week.

Any updates on this?

**Comment by twaddington** [comment]
> > [@dzienisz](https://github.com/dzienisz) I'm chipping away at it. I should have a draft up this week.
> 
> Any updates on this?

Got tied up with other stuff but plan to circle back to this soon.

**Comment by dgottstein** [comment]
Hi,

Is there an update on this issue? I can‘t set fixed coordinates in any way.
I tried CLI via Serial (NoneType error), CLI via TCP (no error but position is not set), WebUI and iPhone App (no input fields available)…

Latest stable firmware, CLI and App-versions.
HW is Heltec Lora v3.2

Maybe there is at least a workaround?

Thanks

**Comment by dzienisz** [comment]
I work here to do it #909. Please give me more time. It's almost done!

**Comment by dzienisz** [comment]
Merged to test
